haystack/indexes.py

```diff
@@ -353,7 +353,12 @@ def load_all_queryset(self):
         By default, returns ``all()`` on the model's default manager.
         """
         return self.get_model()._default_manager.all()
-
+    
+    def pre_process_data(self, queryset):
```

Comment:
Also, should we perhaps make the name self-documenting – `preprocess_queryset` or `preprocess_queryset_slice` to convey that it's not the entire queryset?